llmfoundry/callbacks/generate_callback.py

```diff
@@ -84,9 +84,11 @@ def generate(self, state: State, logger: Logger):
 
         if dist.get_global_rank() == 0:
             if self.wandb_logger is not None:
-                artifact = wandb.Artifact(
-                    'generate_samples_' + str(wandb.run.id),  # type: ignore
-                    type='predictions')
+                assert wandb.run is not None, 'wandb should have started run'
```

Comment:
we generally prefer to raising errors instead of using asserts.

Reply:
same, assert should be ok here

Reply:
`assert` is the right choice here as these are system invariants that should never logically happen. See: https://stefan.sofa-rockers.org/2018/04/16/assertions-and-exceptions/ a good explanation.